From: Suraj Jitindar Singh <surajjs@amazon.com>
Date: Thu, 13 Jun 2024 11:35:42 -0700
Subject: acpi: Support CONFIG_ACPI without CONFIG_PCI

Make is possible to use ACPI without having CONFIG_PCI set.

When initialising ACPI the following call chain occurs:

  acpi_init() ->
    acpi_bus_init() ->
      acpi_load_tables() ->
        acpi_ev_install_region_handlers() ->

acpi_ev_install_region_handlers() calls acpi_ev_install_space_handler() on
each of the default address spaces defined as:

  u8 acpi_gbl_default_address_spaces[ACPI_NUM_DEFAULT_SPACES] = {
          ACPI_ADR_SPACE_SYSTEM_MEMORY,
          ACPI_ADR_SPACE_SYSTEM_IO,
          ACPI_ADR_SPACE_PCI_CONFIG,
          ACPI_ADR_SPACE_DATA_TABLE
  };

However in acpi_ev_install_space_handler() the case statement for
ACPI_ADR_SPACE_PCI_CONFIG is ifdef'd as:

  #ifdef ACPI_PCI_CONFIGURED
                  case ACPI_ADR_SPACE_PCI_CONFIG:

                          handler = acpi_ex_pci_config_space_handler;
                          setup = acpi_ev_pci_config_region_setup;
                          break;
  #endif

ACPI_PCI_CONFIGURED is not defined if CONFIG_PCI is not enabled, thus the
attempt to install the handler fails.

Fix this by ifdef'ing ACPI_ADR_SPACE_PCI_CONFIG in the list of default
address spaces.

Fixes: bd23fac3eaaa ("ACPICA: Remove PCI bits from ACPICA when CONFIG_PCI is unset")
CC: stable@vger.kernel.org # 5.0.x-
Signed-off-by: Suraj Jitindar Singh <surajjs@amazon.com>
---
 drivers/acpi/acpica/evhandler.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/acpi/acpica/evhandler.c b/drivers/acpi/acpica/evhandler.c
--- a/drivers/acpi/acpica/evhandler.c
+++ b/drivers/acpi/acpica/evhandler.c
@@ -26,7 +26,9 @@ acpi_ev_install_handler(acpi_handle obj_handle,
 u8 acpi_gbl_default_address_spaces[ACPI_NUM_DEFAULT_SPACES] = {
 	ACPI_ADR_SPACE_SYSTEM_MEMORY,
 	ACPI_ADR_SPACE_SYSTEM_IO,
+#ifdef ACPI_PCI_CONFIGURED
 	ACPI_ADR_SPACE_PCI_CONFIG,
+#endif
 	ACPI_ADR_SPACE_DATA_TABLE
 };
 

